WalletWasabi/KeyManagement/HdPubKey.cs

```diff
@@ -108,7 +108,7 @@ public void SetKeyState(KeyState state, KeyManager kmToFile = null)
 
 		public bool HasLabel { get; private set; }
 
-		public BitcoinPubKeyAddress GetP2pkhAddress(Network network) => PubKey.GetAddress(network);
+		public BitcoinPubKeyAddress GetP2pkhAddress(Network network) => (BitcoinPubKeyAddress)PubKey.GetAddress(ScriptPubKeyType.Legacy, network);
```

Comment:
That method has an ugly interface. You can use this one:
```suggestion
		public BitcoinPubKeyAddress GetP2pkhAddress(Network network) => (BitcoinPubKeyAddress)PubKey.GetSegwitAddress(network);
```

Reply:
Btw, you changed the type of address. There you have to use a P2WPKH (segwit) instead of a P2PKH (legacy). Accept my proposal or change the enumeration to use segwit addresses.

Reply:
Forget it, your change is okay.

Reply:
It is not working like this.

Reply:
Error	CS0030	Cannot convert type 'NBitcoin.BitcoinWitPubKeyAddress' to 'NBitcoin.BitcoinPubKeyAddress'	WalletWasabi	C:\WORK\WalletWasabi\WalletWasabi\KeyManagement\HdPubKey.cs	111

Reply:
yes, that's why I said _forget it_